design-documents/testing/functional/versioning-and-backward-compatibility-policy.md

```diff
@@ -28,82 +31,75 @@ An approach of defining what each release should include was taken from [Semanti
-1. Add Semantic Version analyzer to be able automatically define release type of MFTF tests package.
+
+1. Add Semantic Version analyzer to be able automatically define the release type of the MFTF tests package.
 2. Update publication infrastructure to exclude tests from `magento2-module` package type.
 3. Introduce publication functionality for publishing `magento2-test-module` package type.
-4. Create metapackage with test packages only for each Magento edition.
+4. Create a metapackage with test packages specifically for each Magento edition.
 
-## Entity's attributes
-Changing any of this attribute will cause Backward Incompatible change.
-
-**Test entity:**
-  
-  |xPath|attribute|
-  |---|---|
-  |`/tests/test`|name|
-  |`/tests/test/<ACTION> ⃰`|stepKey|
-  |`/tests/test/before/<ACTION> ⃰`|stepKey|
-  |`/tests/test/after/<ACTION> ⃰`|stepKey| 
-
-**Action Group entity:**
-
-  |xPath|attribute|
-  |---|---|
-  |`/actionGroups/actionGroup`|name|
-  |`/actionGroups/actionGroup/arguments/argument`|name|
-  |`/actionGroups/actionGroup/arguments/argument`|type|
-  |`/actionGroups/actionGroup/<ACTION> ⃰`|stepKey|
-
-**Data entity:**
-  
-  |xPath|attribute|
-  |---|---|
-  |`/entities/entitie`|name|
-  |`/entities/entitie/data`|key|
-  |`/entities/entitie/array`|key|
-  |`/entities/entitie/var`|key|
-  |`/entities/entitie/requiredEntity`|type|
-
-**Metadata entity:**
-  
-  |xPath|attribute|
-  |---|---|
-  |`/operations/operation`|name|
-  |`/operations/operation/field`|key|
-  |`/operations/operation(/object)+`|key|
-  |`/operations/operation(/object)+/field`|key|
-  |`/operations/operation(/object)+/array`|key|
-  |`/operations/operation/array`|key|
-  |`/operations/operation/array/object`|key|
+## Version increase matrix
   
-**Page entity:**
-
-  |xPath|attribute|
-  |---|---|
-  |`/pages/page`|name|
-  |`/pages/page/section`|name|
-  
-**Section entity:**
-
-  |xPath|attribute|
-  |---|---|
-  |`/sections/section`|name|
-  |`/sections/section/element`|name|
+|Entity Type|Change|Version Increase|
+|---|---|---|
+|ActionGroup|`<actionGroup>` added|MINOR
+| |`<actionGroup>` removed|MAJOR
+| |`<actionGroup>` `<action>` added|MINOR
+| |`<actionGroup>` `<action>` removed|MAJOR
+| |`<actionGroup>` `<action>` type changed|PATCH
+| |`<actionGroup>` `<action>` attribute changed|PATCH
+| |`<actionGroup>` `<argument>` added|MAJOR
+| |`<actionGroup>` `<argument>` removed|MAJOR
+| |`<actionGroup>` `<argument>` changed|MAJOR
+|Data|`<entity>` added|MINOR
+| |`<entity>` removed|MAJOR
+| |`<entity>` `<array>` added|MINOR
+| |`<entity>` `<array>` removed|MAJOR
+| |`<entity>` `<array>` `<item>` removed|PATCH
+| |`<entity>` `<field>` added|MINOR
+| |`<entity>` `<field>` removed|MAJOR
+| |`<entity>` `<required-entity>` added|MAJOR
+| |`<entity>` `<required-entity>` removed|MAJOR
+| |`<entity>` `<var>` added|MAJOR
+| |`<entity>` `<var>` removed|MAJOR
+|Metadata|`<operation>` added|MINOR
+| |`<operation>` removed|MAJOR
+| |`<operation>` changed|PATCH
```

Comment:
This may be MINOR, since compatibility with Magento is changing.
Same relates to `section=>element change`.

Reply:
Agree on this account.

On Section/element changed I realize need to break it into separate categories:
`selector` => `PATCH`
`parameterized` => `MAJOR` (making a selector parameterized will cause existing invocations to break)
`type` => `PATCH`